### ISSUE TITLE
Re-reading old editions of DEMO

### ISSUE THREAD

**Issue by Pieter1024** [jonge-democraten/website#23]
Embedding ISSUU external code.

**Comment by Pieter1024** [comment]
I would propose we do not add the ISSUU external code to our new website. See current implementation at http://jongedemocraten.nl/jdmedia/demo.html.

Reasons for suggesting this:

* Reading DEMO is awkward this way. E.g. ISSUU does not display/function well on tablets.
* Reader privacy: right now, to read DEMO you need to use ISSUU. Therefore, ISSUU knows which articles you have read.
* Visitor privacy: the embedded ISSUU code contains an extra tracker mechanism (Quantcast), which runs whenever someone visits the website. This violates our cookie agreement mechanism and therefore the cookie law.
* Open availability of the magazine: To download DEMO and read it offline, you need to login on ISSUU.
* Embedding JavaScript code in a Content Manager-editable field is a security risk. Whitelisting would be a possible way to avoid this risk.

I would propose to replace the page by a Document Listing and to upload PDFs of all editions of DEMO to the Media Library. In this way, all issues mentioned above are immediately resolved.

I close this issue because I feel it is resolved in this way. I welcome @bashazeborg 's. take on this matter.

**Comment by bashazeborg** [comment]
I agree on Pieter's arguments considering ISSUU and therefore I would say we do not add ISSUU code. It would be great though to make the documents visable for the visitors like our Statue (which is in Scribd now). http://jongedemocraten.nl/organisatie/statuten-en-hr.html

For that reason I move this Issue to Milestone 5.

**Comment by Pieter1024** [comment]
Embedding of documents is now possible. However, it does not work yet in TinyMCE - not sure why.

**Comment by Pieter1024** [comment]
Now, it works.

To embed a PDF in a page:
* Upload the PDF in the media library.
* Open a page or post for editing.
* On the editor toolbar, click the HTML button (seventh from the left).
* In the HTML window, insert the appropriate \<object\> tag:
```
<object data="/static/media/path/to.pdf" width="100%" height=600 type="application/pdf">Text to display if browser does not support PDF embedding</object>
```

The path to the PDF must be local, and start with /static/media/. You can find the appropriate tag in the address bar when opening the uploaded PDF. Width should be 100% to use the full width of the message. Height is in pixels, and should be set appropriately. 600 seems like a good default. The text between the \<object\> and \</object\> tags is alternative text that is displayed if the browser does not support PDF embedding.

Obviously, this is not someting that is easy enough for every content manager to apply. I think the need will only arise in very specific scenarios. Therefore, I feel this approach is sufficient to close the issue.

A demo is available in master since commit 5c04fd0f767f9a3a9428692f12a0b0875b8f84f6. On the Documenten page, one can see an embedded PDF.
